core/sdk-core/src/test/java/utils/HttpTestUtils.java

```diff
@@ -59,9 +76,8 @@ public static SdkClientConfiguration testClientConfiguration() {
                                      .option(SdkAdvancedClientOption.SIGNER, new NoOpSigner())
                                      .option(SdkAdvancedClientOption.USER_AGENT_PREFIX, "")
                                      .option(SdkAdvancedClientOption.USER_AGENT_SUFFIX, "")
-                                     .option(SdkClientOption.ASYNC_RETRY_EXECUTOR_SERVICE, Executors.newScheduledThreadPool(1))
-                                     .option(SdkAdvancedAsyncClientOption.FUTURE_COMPLETION_EXECUTOR,
-                                             Runnable::run)
+                                     .option(SdkClientOption.SCHEDULED_EXECUTOR_SERVICE, Executors.newScheduledThreadPool(1))
```

Comment:
Isnt this thread pool technically unbounded? The core pool size is 1, but the max pool is int.max.

Reply:
I believe the work queue used by the scheduledThreadPool is unbounded. Because the JDK classes only use the core pool size until the work queue is full, and the work queue is unbounded, the maximumPoolSize within the scheduled thread pool should never used.